packages/node_modules/@node-red/editor-client/src/js/ui/view-tools.js

```diff
@@ -120,8 +120,69 @@ RED.view.tools = (function() {
         }
     }
 
+    function setSelectedNodeLabelState(labelShown) {
+        var selection = RED.view.selection();
+        var historyEvents = [];
+        var nodes = [];
+        if (selection.nodes) {
+            selection.nodes.forEach(function(n) {
+                if (n.type !== 'subflow' && n.type !== 'group') {
+                    nodes.push(n);
+                } else if (n.type === 'group') {
+                    nodes = nodes.concat( RED.group.getNodes(n,true));
+                }
+            });
+        }
+        nodes.forEach(function(n) {
+            var modified = false;
+            var oldValue = n.l === undefined?true:n.l;
+            var isLink = /^link (in|out)$/.test(n._def.type);
+
+            if (labelShown) {
+                if (n.l === false || (isLink && !n.hasOwnProperty('l'))) {
```

Comment:
Does this also need a !islink - or maybe do the isLink text first ?

Reply:
If you were watching the live stream you would have seen the discussion around that ;)

The logic is okay here because if `n.l` is precisely `false` then the label is hidden regardless of the node type. The extra `isLink` is needed because the link node's default state (if n.l is `undefined`) is to hide the label